src/setuptools_scm/git.py

```diff
@@ -104,20 +104,39 @@ def fail_on_shallow(wd):
         )
 
 
+def get_working_directory(config):
+    """
+    Return the working directory (``GitWorkdir``).
+    """
+
+    if config.parent:
+        return GitWorkdir.from_potential_worktree(config.parent)
+
+    if config.search_parent_directories:
+        return search_parent(config.absolute_root)
+
+    wd = GitWorkdir.from_potential_worktree(config.absolute_root)
+
+    if wd is not None:
+        return wd
+
+    from .hg_git import GitWorkdirHgClient
+
+    return GitWorkdirHgClient.from_potential_worktree(config.absolute_root)
```

Comment:
@RonnyPfannschmidt 

Concerning the failing CI : in one of the final commits I made I added this function under pressure from *flake8*. I'm quite convinced that the old implementation is unchanged, but it would be good to double check.